src/kaocha/plugin.clj

```diff
@@ -16,22 +16,48 @@
     (catch java.io.FileNotFoundException e
       false)))
 
+(defn- try-require-info [n]
+  [(try-require n) n])
+
 (defn try-load-third-party-lib [plugin-name]
-  (if (qualified-keyword? plugin-name)
-    (when-not (try-require (symbol (str (namespace plugin-name) "." (name plugin-name))))
-      (try-require (symbol (namespace plugin-name))))
-    (try-require (symbol (name plugin-name)))))
+  (let [full-name (symbol (str (namespace plugin-name) "." (name plugin-name)))]
+    (if (qualified-keyword? plugin-name)
+      (if-let [full-result (try-require full-name)]
+        [[full-result full-name]]
+        [[false full-name] (try-require-info (symbol (namespace plugin-name))) ])
+      [(try-require-info (symbol (name plugin-name)))])))
```

Comment:
Could you add a doc string explaining what the inputs/outputs of this function are? I'm having a hard time seeing how to interpret the nested vectors.

Reply:
Hmm, I don't know if it's enough to add a docstring and maybe some names via `let` or if I should also change the representation from vectors.

Right now, successfully loading our hook plugin's namespace would be represented as `[true :kaocha.plugins.hooks]` but I could see that being represented as `{:result true :ns :kaocha.plugins.hooks}`. The second would be way nice for further changes, which might be reason enough.

Reply:
I held off on the docstring, but I did a small refactoring here because I think it will give us a better sense of whether a slight refactoring and better comments is enough or if the hash-map representation is better.